lib/model/list_model.dart

```diff
@@ -49,6 +54,9 @@ class ListModel {
   Future<void> update(Item item) async {
     if (items.contains(item)) {
       await DatabaseManager.putItem(item);
+      // The following ensures that the copy of `item` that `this` has is up to date.
+      items.remove(item);
+      items.add(item);
```

Comment:
This introduces a new bug: when a user updates an item, that item gets moved to the end of the list. You can see the problem occur in the UI by updating a few items, and then adding a new item. Suddenly a bunch of blank spots show up in the UI.  

To maintain order, try something like this (might need modifications):
```suggestion
      // The following ensures that the copy of `item` that `this` has is up to date.
      const index = items.indexOf(item);
      items[index] = item;
```

Reply:
Okay, I think I found a fix. What I ended up doing was calling `items.lookup(item)` to get the `Item` object stored in the links and then copying all the fields of `item` (except `id`) onto that object.